rust/connlib/clients/shared/src/eventloop.rs

```diff
@@ -74,6 +85,28 @@ where
                 Poll::Pending => {}
             }
 
+            if let Poll::Ready(Some(dns)) = self.dns_updated.poll_recv(cx) {
+                self.dns_update = Some(
+                    async move {
+                        // We sleep here to debounce since if another event is generated within the given time frame it will replace this future.
+                        tokio::time::sleep(Duration::from_millis(100)).await;
```

Comment:
100 ms is picked arbitrarily, I'm open to other suggestions.

Reply:
100 might be a little too eager -- I think the flurry of events on Apple spans about 500ms typically. See #4202